\doxysection{werkzeug.\+exceptions Namespace Reference}
\hypertarget{namespacewerkzeug_1_1exceptions}{}\label{namespacewerkzeug_1_1exceptions}\index{werkzeug.exceptions@{werkzeug.exceptions}}
\doxysubsubsection*{Classes}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1___retry_after}{\+\_\+\+Retry\+After}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_aborter}{Aborter}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_bad_gateway}{Bad\+Gateway}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_bad_host}{Bad\+Host}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_bad_request}{Bad\+Request}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_bad_request_key_error}{Bad\+Request\+Key\+Error}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_client_disconnected}{Client\+Disconnected}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_conflict}{Conflict}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_expectation_failed}{Expectation\+Failed}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_failed_dependency}{Failed\+Dependency}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_forbidden}{Forbidden}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_gateway_timeout}{Gateway\+Timeout}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_gone}{Gone}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_h_t_t_p_exception}{HTTPException}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_h_t_t_p_version_not_supported}{HTTPVersion\+Not\+Supported}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_im_a_teapot}{Im\+ATeapot}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_internal_server_error}{Internal\+Server\+Error}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_length_required}{Length\+Required}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_locked}{Locked}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_method_not_allowed}{Method\+Not\+Allowed}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_not_acceptable}{Not\+Acceptable}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_not_found}{Not\+Found}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_not_implemented}{Not\+Implemented}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_precondition_failed}{Precondition\+Failed}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_precondition_required}{Precondition\+Required}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_requested_range_not_satisfiable}{Requested\+Range\+Not\+Satisfiable}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_request_entity_too_large}{Request\+Entity\+Too\+Large}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_request_header_fields_too_large}{Request\+Header\+Fields\+Too\+Large}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_request_timeout}{Request\+Timeout}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_request_u_r_i_too_large}{Request\+URIToo\+Large}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_security_error}{Security\+Error}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_service_unavailable}{Service\+Unavailable}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_too_many_requests}{Too\+Many\+Requests}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_unauthorized}{Unauthorized}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_unavailable_for_legal_reasons}{Unavailable\+For\+Legal\+Reasons}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_unprocessable_entity}{Unprocessable\+Entity}}
\item 
class \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_unsupported_media_type}{Unsupported\+Media\+Type}}
\end{DoxyCompactItemize}
\doxysubsubsection*{Functions}
\begin{DoxyCompactItemize}
\item 
\Hypertarget{namespacewerkzeug_1_1exceptions_a9b297e4e323e146bfd4c25e54ddae624}\label{namespacewerkzeug_1_1exceptions_a9b297e4e323e146bfd4c25e54ddae624} 
None {\bfseries \+\_\+find\+\_\+exceptions} ()
\item 
"{}te.\+No\+Return"{} \mbox{\hyperlink{namespacewerkzeug_1_1exceptions_aebbca130bf48b07f939e7e92f3c6a0c5}{abort}} (t.\+Union\mbox{[}int, "{}Response"{}\mbox{]} status, \texorpdfstring{$\ast$}{*}t.\+Any args, \texorpdfstring{$\ast$}{*}\texorpdfstring{$\ast$}{*}t.\+Any kwargs)
\end{DoxyCompactItemize}
\doxysubsubsection*{Variables}
\begin{DoxyCompactItemize}
\item 
\Hypertarget{namespacewerkzeug_1_1exceptions_ac9608c1309325e45d03776ca4dfdc587}\label{namespacewerkzeug_1_1exceptions_ac9608c1309325e45d03776ca4dfdc587} 
\mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_aborter}{Aborter}} {\bfseries \+\_\+aborter} = \mbox{\hyperlink{classwerkzeug_1_1exceptions_1_1_aborter}{Aborter}}()
\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
\begin{DoxyVerb}Implements a number of Python exceptions which can be raised from within
a view to trigger a standard HTTP non-200 response.

Usage Example
-------------

.. code-block:: python

    from werkzeug.wrappers.request import Request
    from werkzeug.exceptions import HTTPException, NotFound

    def view(request):
        raise NotFound()

    @Request.application
    def application(request):
        try:
            return view(request)
        except HTTPException as e:
            return e

As you can see from this example those exceptions are callable WSGI
applications. However, they are not Werkzeug response objects. You
can get a response object by calling ``get_response()`` on a HTTP
exception.

Keep in mind that you may have to pass an environ (WSGI) or scope
(ASGI) to ``get_response()`` because some errors fetch additional
information relating to the request.

If you want to hook in a different exception page to say, a 404 status
code, you can add a second except for a specific subclass of an error:

.. code-block:: python

    @Request.application
    def application(request):
        try:
            return view(request)
        except NotFound as e:
            return not_found(request)
        except HTTPException as e:
            return e\end{DoxyVerb}
 

\doxysubsection{Function Documentation}
\Hypertarget{namespacewerkzeug_1_1exceptions_aebbca130bf48b07f939e7e92f3c6a0c5}\index{werkzeug.exceptions@{werkzeug.exceptions}!abort@{abort}}
\index{abort@{abort}!werkzeug.exceptions@{werkzeug.exceptions}}
\doxysubsubsection{\texorpdfstring{abort()}{abort()}}
{\footnotesize\ttfamily \label{namespacewerkzeug_1_1exceptions_aebbca130bf48b07f939e7e92f3c6a0c5} 
 "{}te.\+No\+Return"{} werkzeug.\+exceptions.\+abort (\begin{DoxyParamCaption}\item[{t.\+Union\mbox{[}int, "{}Response"{}\mbox{]}}]{status}{, }\item[{\texorpdfstring{$\ast$}{*}t.\+Any}]{args}{, }\item[{\texorpdfstring{$\ast$}{*}\texorpdfstring{$\ast$}{*}t.\+Any }]{kwargs}{}\end{DoxyParamCaption})}

\begin{DoxyVerb}Raises an :py:exc:`HTTPException` for the given status code or WSGI
application.

If a status code is given, it will be looked up in the list of
exceptions and will raise that exception.  If passed a WSGI application,
it will wrap it in a proxy WSGI exception and raise that::

   abort(404)  # 404 Not Found
   abort(Response('Hello World'))\end{DoxyVerb}
 